x-pack/plugins/cases/server/services/cases/index.test.ts

```diff
@@ -0,0 +1,1160 @@
+          .calls[0][2] as Partial<ESCaseAttributes>;
+        expect(connector).toMatchInlineSnapshot(`
+          Object {
+            "fields": Array [
+              Object {
+                "key": "issueType",
+                "value": "bug",
+              },
+              Object {
+                "key": "priority",
+                "value": "high",
+              },
+              Object {
+                "key": "parent",
+                "value": "2",
+              },
+            ],
+            "name": ".jira",
+            "type": ".jira",
+          }
+        `);
+      });
+
+      it('removes the connector id and adds it to the references', async () => {
+        unsecuredSavedObjectsClient.update.mockReturnValue(
+          Promise.resolve({} as SavedObjectsUpdateResponse<ESCaseAttributes>)
+        );
+
+        await service.patchCase({
+          caseId: '1',
+          unsecuredSavedObjectsClient,
+          updatedAttributes: createCaseUpdateParams(createJiraConnector()),
+          originalCase: {} as SavedObject<CaseAttributes>,
+        });
+
+        const updateAttributes = unsecuredSavedObjectsClient.update.mock
+          .calls[0][2] as Partial<ESCaseAttributes>;
+        expect(updateAttributes.connector).not.toHaveProperty('id');
+
+        const updateOptions = unsecuredSavedObjectsClient.update.mock
+          .calls[0][3] as SavedObjectsUpdateOptions<unknown>;
+        expect(updateOptions.references).toMatchInlineSnapshot(`
+          Array [
+            Object {
+              "id": "1",
+              "name": "connectorId",
+              "type": "action",
+            },
+          ]
+        `);
+      });
+
+      it('removes the external_service connector_id and adds it to the references', async () => {
+        unsecuredSavedObjectsClient.update.mockReturnValue(
+          Promise.resolve({} as SavedObjectsUpdateResponse<ESCaseAttributes>)
+        );
+
+        await service.patchCase({
+          caseId: '1',
+          unsecuredSavedObjectsClient,
+          updatedAttributes: createCasePostParams(getNoneCaseConnector(), createExternalService()),
+          originalCase: {} as SavedObject<CaseAttributes>,
+        });
+
+        const updateAttributes = unsecuredSavedObjectsClient.update.mock
+          .calls[0][2] as Partial<ESCaseAttributes>;
+        expect(updateAttributes.external_service).not.toHaveProperty('connector_id');
+
+        const updateOptions = unsecuredSavedObjectsClient.update.mock
+          .calls[0][3] as SavedObjectsUpdateOptions<unknown>;
+        expect(updateOptions.references).toMatchInlineSnapshot(`
+          Array [
+            Object {
+              "id": "100",
+              "name": "pushConnectorId",
+              "type": "action",
+            },
+          ]
+        `);
+      });
+
+      it('builds references for connector_id, connector.id, and includes the existing references', async () => {
```

Comment:
nit: `for connector_id, connector.id` -> `for external service connector id, case connector id`